flinkfood-demo/src/main/java/org/flinkfood/flinkjobs/DishViewJob.java

```diff
@@ -0,0 +1,86 @@
+import org.apache.flink.table.api.*;
+import org.apache.flink.types.Row;
+import static org.apache.flink.table.api.Expressions.*;
+import org.apache.flink.table.api.bridge.java.StreamTableEnvironment;
+import org.flinkfood.schemas.*;
+
+// Class declaration for the Flink job
+public class DishViewJob {
+
+        // Kafka and MongoDB connection details obtained from environment variables
+        private static final String KAFKA_URI = System.getenv("KAFKA_URI");
+        private static final String SOURCE_DISH_TABLE = "postgres.public.dishes";
+        private static final String SOURCE_RESTAURANT_INFO_TABLE = "postgres.public.restaurant_info";
+
+        // Main method where the Flink job is defined
+        public static void main(String[] args) throws Exception {
+
+                // Setting up Kafka source with relevant configurations
+                KafkaSource<Dish> dishSource = KafkaSource.<Dish>builder()
+                                .setBootstrapServers(KAFKA_URI)
+                                .setTopics(SOURCE_DISH_TABLE)
+                                .setGroupId("my-group")
+                                .setStartingOffsets(OffsetsInitializer.earliest())
+                                .setValueOnlyDeserializer(new KafkaDishSchema())
+                                .build();
+
+                // Setting up Kafka source with relevant configurations
+                KafkaSource<RestaurantInfo> restaurantInfoSource = KafkaSource.<RestaurantInfo>builder()
+                                .setBootstrapServers(KAFKA_URI)
+                                .setTopics(SOURCE_RESTAURANT_INFO_TABLE)
+                                .setGroupId("my-group")
+                                .setStartingOffsets(OffsetsInitializer.earliest())
+                                .setValueOnlyDeserializer(new KafkaRestaurantInfoSchema())
+                                .build();
+
+                // Setting up Flink execution environment
+                StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+                StreamTableEnvironment tableEnv = StreamTableEnvironment.create(env);
+
+                // Creates a Flink data stream for the dishes
+                DataStream<Dish> dishStream = env
+                .fromSource(dishSource, WatermarkStrategy.noWatermarks(), "Kafka Dish Source")
+                .setParallelism(1);
+                
+                // Creates a Flink data stream for the restaurants
+                DataStream<RestaurantInfo> restaurantInfoStream = env
+                                .fromSource(restaurantInfoSource, WatermarkStrategy.noWatermarks(),
+                                                "Kafka Restaurant Info Source")
+                                .setParallelism(1);
+
+                // Creates a Flink table for the dishes
+                Table dishTable = tableEnv.fromDataStream(dishStream).select(
```

Comment:
I would think back about the commutation from DataStream to Tables, and back, I think could be inefficient